src/regex/engine.cr

```diff
@@ -1,4 +1,8 @@
-{% if flag?(:use_pcre2) %}
+# The following condition ensures that the engine selection respects `-Duse_pcre2`/`-Duse_pcre`,
+# and if none is given it tries to check for availability of `libpcre2` with `pkg-config`.
+# If `pkg-config` is unavailable, the default is PCRE2. If `pkg-config` is available but
+# has no information about a `libpcre2` package, it falls back to PCRE.
+{% if flag?(:use_pcre2) || (!flag?(:use_pcre) && (flag?(:win32) || `hash pkg-config 2> /dev/null && (pkg-config --silence-errors --modversion libpcre2-8 || printf %s false) || true` != "false")) %}
```

Comment:
If you run `make crystal FLAGS=-Duse_pcre2` then both flags will be present and it is PCRE2 that will be chosen. It seems this is by design, contrary to my suggestion [here](https://github.com/crystal-lang/crystal/pull/12840#discussion_r1048386259). Documenting that behavior would be nice even if these flags are temporary

Reply:
The two steps to remedy this is to err if the two flags are present, and fix the makefile to not add the flag if there's one of these there, right?

Reply:
You'd have to test for both `-Duse_pcre2` and `-D use_pcre2` with the space inside the Makefile, which sounds rather complex for such a small task, and that's not counting `CRYSTAL_OPTS`, so perhaps this behavior is fine. It is not like those flags would exist for more than 3 minor releases or something